imports/client/components/FirehosePage.tsx

```diff
@@ -0,0 +1,267 @@
+import { Meteor } from 'meteor/meteor';
+import { useTracker } from 'meteor/react-meteor-data';
+import { _ } from 'meteor/underscore';
+import { faEraser } from '@fortawesome/free-solid-svg-icons/faEraser';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import React, {
+  useCallback, useEffect, useMemo, useRef, useState,
+} from 'react';
+import Button from 'react-bootstrap/Button';
+import FormControl, { FormControlProps } from 'react-bootstrap/FormControl';
+import FormGroup from 'react-bootstrap/FormGroup';
+import InputGroup from 'react-bootstrap/InputGroup';
+import { RouteComponentProps } from 'react-router';
+import { Link } from 'react-router-dom';
+import styled from 'styled-components';
+import { shortCalendarTimeFormat } from '../../lib/calendarTimeFormat';
+import ChatMessages from '../../lib/models/chats';
+import Profiles from '../../lib/models/profiles';
+import Puzzles from '../../lib/models/puzzles';
+import { ChatMessageType } from '../../lib/schemas/chat';
+import { PuzzleType } from '../../lib/schemas/puzzle';
+import { useBreadcrumb } from '../hooks/breadcrumb';
+import FixedLayout from './styling/FixedLayout';
+
+const FirehosePageLayout = styled.div`
+  padding: 8px 15px;
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+  justify-content: flex-start;
+  max-height: calc(100vh - 60px);
+  > * {
+    width: 100%;
+  }
+`;
+
+interface MessageProps {
+  msg: ChatMessageType;
+  displayNames: Record<string, string>;
+  puzzles: Record<string, PuzzleType>;
+}
+
+const Message = ({ msg, displayNames, puzzles }: MessageProps) => {
+  const displayName = msg.sender ? displayNames[msg.sender] : 'jolly-roger';
+  const ts = shortCalendarTimeFormat(msg.timestamp);
+  return (
+    <div>
+      [
+      {ts}
+      ] [
+      <Link to={`/hunts/${msg.hunt}/puzzles/${msg.puzzle}`}>{puzzles[msg.puzzle].title}</Link>
+      {'] '}
+      {displayName}
+      {': '}
+      {msg.text}
+    </div>
+  );
+};
+
+const MessagesPane = styled.div`
+  overflow-y: scroll;
+  max-height: 100vh;
```

Comment:
Is this necessary? I don't think it does anything.
```suggestion
```

Reply:
You're right.  I incorporated the patch in the rebase.